extensions/tn_cache/internal/db_ops.go

```diff
@@ -421,20 +340,28 @@ func (c *CacheDB) CacheEventsWithIndex(ctx context.Context, events []CachedEvent
 					value = EXCLUDED.value
 			`, dataProviders, streamIDs, eventTimes, indexValues)
 
-			if err != nil {
-				return nil, fmt.Errorf("insert index events batch: %w", err)
+					if err != nil {
+						return nil, fmt.Errorf("insert index events batch: %w", err)
+					}
+				}
 			}
-		}
-	}
 
-			// Update the last refreshed timestamp for the stream
+			// Update the last refreshed timestamp and height for the stream
 			if primaryProvider != "" && primaryStreamID != "" {
 				now := time.Now().Unix()
-				_, err = tx.Execute(traceCtx, `
-					UPDATE `+constants.CacheSchemaName+`.cached_streams
-					SET last_refreshed = $3
-					WHERE data_provider = $1 AND stream_id = $2
-				`, primaryProvider, primaryStreamID, now)
+				if height != nil {
```

Comment:
I didn't understand what would mean to have a nil height here, is it still safe? shouldn't it always be defined when updating?